js/src/threepio.js

```diff
@@ -10,39 +11,34 @@ export default class Threepio {
     this.tf = tf;
   }
 
-  normalizeFunctionName(name) {
-    return name
+  _normalizeFunctionName(name, toLang) {
+    const result = name
       .match(NORMALIZATION_REGEX)
       .join('')
       .toLowerCase();
-  }
 
-  matchArgs(args, argMap) {
-    // Sort args by index
-    argMap.args.sort((a, b) =>
-      a.index === b.index ? 0 : a.index < b.index ? -1 : 1
-    );
-    for (const [i, arg] of args.entries()) {
-      // TODO: match args based on fromLibrary name
-      // TODO: Ensure order is preserved
-      // TODO: Ensure there are not any missing required arguments
-      console.log(i, arg);
-      continue;
+    if (result in this.mappedCommands[toLang]) {
+      return result;
+    }
+
+    if (result in WORD_TRANSLATIONS) {
+      return WORD_TRANSLATIONS[result];
     }
 
-    return args;
+    throw new Error(translationMissing(name));
   }
 
   translate(toLang, cmd) {
     const cmdInfo = this.mappedCommands[toLang][
-      this.normalizeFunctionName(cmd.functionName)
+      this._normalizeFunctionName(cmd.functionName, toLang)
     ];
 
     let translatedCmd = this[cmdInfo.attrs.shift()];
     while (cmdInfo.attrs.length > 0) {
       translatedCmd = translatedCmd[cmdInfo.attrs.shift()];
     }
 
-    translatedCmd(...cmd.args);
+    cmd.execFn = translatedCmd;
```

Comment:
Hmm, has side-effect on cmd outside of this method, is that intended?
(I.e. returning cmd is not necessary in this case)